src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -129,12 +133,33 @@ public synchronized void updateSubscriptions(Set<String> topics, @NonNull Consum
         topicsToSubscribe.removeAll(subscribedIotCoreTopics);
 
         topicsToSubscribe.forEach(s -> {
-            try {
-                subscribeToIotCore(s);
-                LOGGER.atDebug().kv(TOPIC, s).log("Subscribed to topic");
-                subscribedIotCoreTopics.add(s);
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
-                LOGGER.atError().kv(TOPIC, s).log("Failed to subscribe");
+            while (!Thread.currentThread().isInterrupted()) {
```

Comment:
why handle this differently than the other mqtt client?

Reply:
`RetryUtils.runWithRetry()`, though concise, takes in specific exception to retry and does not consider _cause_ of the exception. I didn't want to retry `ExceutionException` thrown by IoTCoreClient.subscribe() if it was caused due to `InterruptedException`

Reply:
You can pull out the exception and rethrow as needed.

Reply:
Makes sense, though we will have to add a custom backoff logic for re-trying pulled out `ExecutionException` when not caused by `InterruptedException`.
```
while(true) {
	try {
		retryUtils.runWithRetry(..., subscribeToIoTCore(), ...);
                return;
	} catch (ExceutionException e) {
		Throwable cause = e.getCause();
		if (cause instanceOf InterruptedException) {
			return;  // or rethrow InterruptedException
		}
		// add a backoff here before retrying
	}
}
```
This makes RetryUtils.runWithRetry() _not well-suited_ here; having the custom retry mechanism makes more sense to me.

Reply:
No, you don't need that. Read the exception and rethrow _inside_ the runWithRetry lambda.

Reply:
thanks, this makes sense. Updated.